src/app/admin/modules/data-management/components/visual-graph-builder/visual-graph-builder.component.ts

```diff
@@ -0,0 +1,222 @@
+        },
+        z: 3,
+        label: {
+            normal: {
+                position: 'top',
+                show: true
+            }
+        },
+        data: this.seriesData[0].data
+    }, {
+        name: '参加培训人数',
+        type: 'bar',
+        itemStyle: {
+            normal: {
+                color: '#e5323e'
+            }
+        },
+        z: 3,
+        label: {
+            normal: {
+                position: 'top',
+                show: true
+            }
+        },
+        data: this.seriesData[1].data
+    }]
+  };
+
+  barChartOption: EChartOption = {
+    legend: {
+        data: ['总人数', '培训人数']
+    },
+    grid: {
+        left: '3%',
+        right: '4%',
+        bottom: '3%',
+        containLabel: true
+    },
+    yAxis:  {
+        type: 'value',
+        data: ['asd', 'qwe', '111', '3', '4', '12344', '555']
+    },
+    xAxis: {
+        type: 'category',
+        data: this.xAxisList
+    },
+    tooltip: {
+        trigger: 'axis',
+        formatter:  (c) => {
+            return Math.round(c[0].value / c[1].value * 100) + '%';
+        }
+    },
+    series: [
+        {
+            name: '培训人数',
+            type: 'bar',
+            stack: '1',
+            label: {
+                normal: {
+                    show: true,
+                    position: 'insideTop'
+                }
+            },
+            data: this.seriesData[0].data
+        },
+        {
+            name: '总人数',
+            type: 'bar',
+            stack: '1',
+            label: {
+                normal: {
+                    show: true,
+                    position: 'insideTop'
+                }
+            },
+            data: this.seriesData[1].data
+        }
+    ]
+  };
+  pieChartOption: EChartOption = {
+    title: [{
+        text: this.title,
+        left: 'center',
+        top: 20,
+        textStyle: {
+            color: '#cccc'
+        }
+    }],
+
+    tooltip: {
+        trigger: 'item',
+        formatter: '{a} <br/>{b} : {c} ({d}%)'
+    },
+
+    visualMap: [{
+        show: false,
+        min: 80,
+        max: 600,
+        inRange: {
+            colorLightness: [0, 1]
+        }
+    }],
+    series: [
+        {
+            name: '访问来源',
+            type: 'pie',
+            radius: '55%',
+            center: ['50%', '50%'],
+            data: [],
+            roseType: 'radius',
+
+            animationType: 'scale',
+            animationEasing: 'elasticOut',
+            animationDelay:  () => {
+                return Math.random() * 200;
+            }
+        }
+    ]
+  };
+
+  @Input() set graphParam(val: ParamSelector) {
+    if (!(val && Object.keys(val)))return;
+    this.titleYear = val.selectedStartYear === val.selectedEndYear ?
+        `${val.selectedStartYear}` : `${val.selectedStartYear}-${val.selectedEndYear}`;
+    this.titleDepartment = val.selectedDepartment;
+    this.title = `${this.titleYear} ${this.titleDepartment} ${this.titleGraphNames}`;
```

Comment:
Correct if I'm wrong:
This component expects `graphParam` and `titleGraphNames` from it's caller, in your logic from it's parent component, the `titleGraphNames` depends on `graphParam`, when `graphParam` changes, the `titleGraphNames` should change accordingly, but can you be sure about that when the `seg graphParam` is called, the `titleGraphNames` of this component is updated? If `titleGraphNames` is not updated, then you will get a wrong `title`. `titleGraphNames` might be changed later, but I doubt that your `set graphParam` will be called again. According to the [offical doc](https://angular.io/guide/component-interaction#intercept-input-property-changes-with-ngonchanges), use `set` here might not be a wise option:
> You may prefer this approach to the property setter when watching multiple, interacting input properties.
Learn about ngOnChanges() in the LifeCycle Hooks chapter.